test/src/nonunity/userobjects/TestNEML2StressUO.C

```diff
@@ -0,0 +1,129 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/*                       BlackBear                              */
+/*                                                              */
+/*           (c) 2017 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#ifdef NEML2_ENABLED
+
+#include "neml2/base/HITParser.h"
+#include "TestNEML2StressUO.h"
+#include "MathUtils.h"
+
+registerMooseObject("BlackBearTestApp", TestNEML2StressUO);
+
+InputParameters
+TestNEML2StressUO::validParams()
+{
+  auto params = TestNEML2StressUOParent::validParams();
+  params.addRequiredParam<MaterialPropertyName>("strain", "The name of strain");
+  params.addRequiredParam<FileName>("input", "Path to input file containing the NEML2 model");
+  params.addRequiredParam<std::string>("model", "Model name in the NEML2 input file");
+  params.addParam<bool>(
+      "verbose",
+      false,
+      "Set to true to print additional information during the batched computation.");
+  return params;
+}
+
+TestNEML2StressUO::TestNEML2StressUO(const InputParameters & params)
+  : TestNEML2StressUOParent(params, "strain"),
+    _fname(getParam<FileName>("input")),
+    _mname(getParam<std::string>("model")),
+    _verbose(getParam<bool>("verbose"))
+{
+  neml2::HITParser parser;
+  parser.parse_and_manufacture(_fname);
+  _model = neml2::Factory::get_object_ptr<neml2::Model>("Models", _mname);
+}
+
+void
+TestNEML2StressUO::batchCompute()
+{
+  neml2::TorchSize nbatch = _input_data.size();
+
+  // First create a torch::Tensor from the Blackbear input data
+  if (_verbose)
+    _console << COLOR_CYAN << name() << ": Converting Blackbear input data to NEML2 format"
+             << COLOR_DEFAULT << std::endl;
+  std::vector<std::array<Real, 6>> neml_strains(nbatch);
+  for (const auto i : index_range(_input_data))
+    neml_strains[i] = R2TtoSymR2T(std::get<0>(_input_data[i]));
+  neml2::LabeledVector in(torch::from_blob(&neml_strains[0][0], {nbatch, 6}, TorchDefaults),
```

Comment:
We really need to get rid of TorchDefaults on our end.

From MOOSE's point of view I think we probably want to leave the type and the host device up to NEML2.  That way we can manage multiple devices on the NEML2 side.

So we may want to wrap these details in a helper over on the NEML2 side.  We want something that takes a blob and returns a blob with the actual tensor construction happening over there.

Reply:
This is done.